hashring/hashring.go

```diff
@@ -389,18 +389,19 @@ func (r *HashRing) lookupNNoLock(key string, n int) []string {
 
 	hash := r.hashfunc(key)
 	unique := make(map[valuetype]struct{})
+	orderedUnique := make([]valuetype, 0, n)
 
 	// lookup N unique servers from the red-black tree. If we have not
 	// collected all the servers we want, we have reached the
 	// end of the red-black tree and we need to loop around and inspect the
 	// tree starting at 0.
-	r.tree.LookupNUniqueAt(n, replicaPoint{hash: hash}, unique)
+	r.tree.LookupNUniqueAt(n, replicaPoint{hash: hash}, unique, &orderedUnique)
```

Comment:
This is an unusual way to work with slices - it's typical to return the modified slice instead of passing pointers. Why pass a ref here?